src/main/java/org/folio/processing/events/services/publisher/KafkaEventPublisher.java

```diff
@@ -72,12 +74,19 @@ public KafkaEventPublisher(KafkaConfig kafkaConfig, Vertx vertx, int maxDistribu
       KafkaProducerRecord<String, String> record =
         KafkaProducerRecord.create(topicName, key, Json.encode(event));
 
-      record.addHeaders(List.of(
-        KafkaHeader.header(OKAPI_URL_HEADER, eventPayload.getOkapiUrl()),
-        KafkaHeader.header(OKAPI_TENANT_HEADER, eventPayload.getTenant()),
-        KafkaHeader.header(OKAPI_TOKEN_HEADER, eventPayload.getToken()),
-        KafkaHeader.header(RECORD_ID_HEADER, recordId),
-        KafkaHeader.header(CHUNK_ID_HEADER, chunkId)));
+      List<KafkaHeader> headers = new ArrayList<>();
+      headers.add(KafkaHeader.header(OKAPI_URL_HEADER, eventPayload.getOkapiUrl()));
+      headers.add(KafkaHeader.header(OKAPI_TENANT_HEADER, eventPayload.getTenant()));
+      headers.add(KafkaHeader.header(OKAPI_TOKEN_HEADER, eventPayload.getToken()));
+      if (recordId != null) {
```

Comment:
Better to check using StringUtils.isNotBlank()

Reply:
@SerhiiNosko  agree, thank you!